AMM/project/contracts/exchange-contract/src/main.sw

```diff
@@ -228,19 +228,17 @@ impl Exchange for Contract {
         let input_asset = msg_asset_id();
         let output_asset = determine_output_asset(input_asset, storage.pair);
 
-        require(deadline >= height(), InputError::DeadlinePassed);
+        require(deadline >= height(), InputError::DeadlinePassed(deadline));
 
         let exact_input = msg_amount();
-        require(exact_input > 0, InputError::AmountCannotBeZero);
+        require(exact_input > 0, InputError::ExpectedNonZeroAmount);
 
         let sender = msg_sender().unwrap();
         let input_asset_in_reserve = storage.reserves.get(input_asset);
         let output_asset_in_reserve = storage.reserves.get(output_asset);
 
         let bought = minimum_output_given_exact_input(exact_input, input_asset_in_reserve, output_asset_in_reserve, LIQUIDITY_MINER_FEE);
 
-        require(bought <= output_asset_in_reserve, TransactionError::InsufficientLiquidity);
```

Comment:
Why was this require removed?

Reply:
`bought` is calculated based on `output_asset_in_reserve`, so that `require` is never violated. This is something I caught thanks to the error messages in revert tests :)